### PR TITLE
[Model][LoRA]LoRA support added for Qwen2VLForConditionalGeneration

### PR DESCRIPTION
**Pull request by ericperfect**
[Model][LoRA]LoRA support added for Qwen2VLForConditionalGeneration
LoRA supported for inference in Qwen2VL
ping @ywang96 @DarkLight1337

### CODE REVIEW

File: vllm/model_executor/models/qwen2_vl.py [hunk 1]
```diff
@@ -928,14 +928,37 @@ def input_processor_for_qwen2_vl(
...
+    }
+
+    # LoRA specific attributes
+    supported_lora_modules = [
+        "qkv_proj",
+        "o_proj",
+        "gate_up_proj",
+        "down_proj",
+    ]
+    embedding_modules = {}
```
Comment (jeejeelee) [inline vllm/model_executor/models/qwen2_vl.py:952]: Although vllm currently only supports lora inference for language model, considering that inference for vision modules will be supported in the future, we should also add vision encoder related modules.

Comment (ericperfect) [reply vllm/model_executor/models/qwen2_vl.py:952]: However, currently fine-tuning only the language model with LoRA achieves good results and meets the needs of most users. Further optimizations can be made later if necessary.

Comment (jeejeelee) [reply vllm/model_executor/models/qwen2_vl.py:952]: Yes, I agree with your view. I just think it might be better to complete the additions first. If you're interested, you can refer to Qwen-VL's implementation, see: https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/models/qwen.py#L1015-L1046.

**Comment by jeejeelee** [comment]
You can run `bash format.sh` to fix the code formatting errors

File: vllm/model_executor/models/qwen2_vl.py (hunk 1 above)
Comment (ericperfect) [reply vllm/model_executor/models/qwen2_vl.py:952]: Thanks, I'll take a look.

File: vllm/model_executor/models/qwen2_vl.py [hunk 2]
```diff
@@ -126,18 +125,15 @@ def __init__(
...
                                         hidden_features,
-                                        quant_config=quant_config,
-                                        prefix=f"{prefix}.fc1")
+                                        quant_config=quant_config)
         self.act = act_layer()
         self.fc2 = RowParallelLinear(hidden_features,
                                      in_features,
-                                     quant_config=quant_config,
-                                     prefix=f"{prefix}.fc2")
+                                     quant_config=quant_config)
```
Comment (jeejeelee) [inline vllm/model_executor/models/qwen2_vl.py:136]: Iseems your force merge caused these unnecessary changes

**Review by jeejeelee — CHANGES_REQUESTED** [review]

**Review by jeejeelee — CHANGES_REQUESTED** [review]
Alternatively, you could add  TODO above the `supported_lora_modules` , indicating that we need to add LoRA for the visual encoder in the future. Additionally, you need to update the documentation to specify that this model supports LoRA.

**Comment by ericperfect** [comment]
Thanks, I already have added the TODO information and updated the documentation.

**Review by DarkLight1337 — APPROVED** [review]
Thanks for adding this!

Also thanks @jeejeelee for the review
